app/controllers/user.js

```diff
@@ -0,0 +1,30 @@
+const bcrypt = require('bcryptjs');
+const User = require('../models').User;
+
+exports.signUp = (req, res, next) => {
+  const user = req.body
+    ? {
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        password: req.body.password,
+        email: req.body.email
+      }
+    : {};
+  bcrypt
+    .hash(user.password, 10)
+    .then(hash => {
+      user.password = hash;
```

Comment:
It would be nice to add a logger event here, stating that we are about to attempt an user creation .You can do that with `logger.info('Some message')` (You will need to add `const logger = require('../logger')` at the beggining of the file)